docker-compose/cassandra-3.11/docker-compose.yml

```diff
@@ -10,34 +10,55 @@ services:
       - HEAP_NEWSIZE=128M
       - MAX_HEAP_SIZE=1024M
       - CASSANDRA_CLUSTER_NAME=c3-${CASSTAG}-cluster
+    healthcheck:
+      test: [ "CMD", "cqlsh", "-u cassandra", "-p cassandra" ,"-e describe keyspaces" ]
+      interval: 15s
+      timeout: 10s
+      retries: 10
+
   cassandra-2:
     image: cassandra:${CASSTAG}
     networks:
       - stargate
     mem_limit: 2G
     depends_on:
-      - cassandra-1
+      cassandra-1:
+        condition: service_healthy
     environment:
       - HEAP_NEWSIZE=128M
       - MAX_HEAP_SIZE=1024M
       - CASSANDRA_SEEDS=cassandra-1
       - CASSANDRA_CLUSTER_NAME=c3-${CASSTAG}-cluster
+    healthcheck:
+      test: [ "CMD", "cqlsh", "-u cassandra", "-p cassandra" ,"-e describe keyspaces" ]
+      interval: 15s
+      timeout: 10s
+      retries: 10
+
   cassandra-3:
     image: cassandra:${CASSTAG}
     networks:
       - stargate
     mem_limit: 2G
     depends_on:
-      - cassandra-2
+      cassandra-2:
```

Comment:
* can't we wait on `cassandra-1` only, would be way faster to startup
* and actually, I think 2 storage nodes are perfectly fine here, why are we adding a third one?

Reply:
I believe the best practice is to add a single node to a cluster at a time which is why `cassandra-3` depends on `cassandra-2`.

I preserved the convention we've traditionally had in docker compose files of having a three node cluster. I don't have a strong opinion about that.